### ISSUE TITLE
 Pruning hang problem in resnet50 imagenet experiment

### ISSUE THREAD

**Issue by xiaomr** [Tencent/PocketFlow#109]
Hi, I try resnet50 imagenet experiment with channel pruning auto mode, however it seems to hang with the following last log,  could you please look at this problem?

INFO:tensorflow:Channel pruning the model/resnet_model/conv2d_52/Conv2D layer,       the pruning rate is 1
INFO:tensorflow:acc_top5: 0.15468749403953552
INFO:tensorflow:accuracy: 0.15468749403953552
INFO:tensorflow:loss: 6.122820854187012
INFO:tensorflow:acc_top1: 0.05494791641831398
INFO:tensorflow:Pruning accuracy 0.15468749403953552
INFO:tensorflow:The current model flops is 737177119.0
INFO:tensorflow:Pruned flops 737177119.0
INFO:tensorflow:The accuracy is 0.15468749403953552 and the flops after pruning is 737177119.0
INFO:tensorflow:The speedup ratio is 0.18036519284567742
INFO:tensorflow:The original model flops is 4087136256.0
INFO:tensorflow:The pruned flops is 737177119.0
INFO:tensorflow:The max strategy dict is {'model/resnet_model/conv2d_46/Conv2D': [0.298828125, 1.0], 'model/resnet_model/conv2d_25/Conv2D': [0.28515625, 1.0], 'model/resnet_model/conv2d_48/Conv2D': [0.2734375, 0.294921875], 'model/resnet_model/conv2d_39/Conv2D': [0.28515625, 1.0], 'model/resnet_model/conv2d_35/Conv2D': [0.26171875, 0.28515625], 'model/resnet_model/conv2d_43/Conv2D': [0.283203125, 1.0], 'model/resnet_model/conv2d_8/Conv2D': [0.34765625, 0.328125], 'model/resnet_model/conv2d_24/Conv2D': [0.271484375, 1.0], 'model/resnet_model/conv2d_51/Conv2D': [0.265625, 1], 'model/resnet_model/conv2d_23/Conv2D': [0.28125, 1.0], 'model/resnet_model/conv2d/Conv2D': [1.0, 0.34375], 'model/resnet_model/conv2d_5/Conv2D': [0.328125, 0.3125], 'model/resnet_model/conv2d_38/Conv2D': [0.26171875, 0.28515625], 'model/resnet_model/conv2d_3/Conv2D': [0.296875, 0.296875], 'model/resnet_model/conv2d_41/Conv2D': [0.26171875, 0.28515625], 'model/resnet_model/conv2d_21/Conv2D': [0.3125, 0.3046875], 'model/resnet_model/conv2d_19/Conv2D': [0.2890625, 0.28125], 'model/resnet_model/conv2d_36/Conv2D': [0.28515625, 1.0], 'model/resnet_model/conv2d_4/Conv2D': [0.296875, 1.0], 'model/resnet_model/conv2d_16/Conv2D': [0.296875, 0.28125], 'model/resnet_model/conv2d_50/Conv2D': [0.328125, 0.265625], 'model/resnet_model/conv2d_45/Conv2D': [0.271484375, 0.298828125], 'model/resnet_model/conv2d_12/Conv2D': [0.33984375, 1.0], 'model/resnet_model/conv2d_47/Conv2D': [0.318359375, 0.2734375], 'model/resnet_model/conv2d_7/Conv2D': [0.3125, 1.0], 'model/resnet_model/conv2d_37/Conv2D': [0.326171875, 0.26171875], 'model/resnet_model/conv2d_6/Conv2D': [0.3125, 0.3125], 'model/resnet_model/conv2d_11/Conv2D': [0.27734375, 1.0], 'model/resnet_model/conv2d_18/Conv2D': [0.30078125, 0.2890625], 'model/resnet_model/conv2d_31/Conv2D': [0.2978515625, 0.26953125], 'model/resnet_model/conv2d_10/Conv2D': [0.328125, 1.0], 'model/resnet_model/conv2d_30/Conv2D': [0.2734375, 1.0], 'model/resnet_model/conv2d_33/Conv2D': [0.2734375, 1.0], 'model/resnet_model/conv2d_13/Conv2D': [0.28125, 0.2734375], 'model/resnet_model/conv2d_44/Conv2D': [0.2783203125, 1.0], 'model/resnet_model/conv2d_49/Conv2D': [0.294921875, 1.0], 'model/resnet_model/conv2d_32/Conv2D': [0.26953125, 0.2734375], 'model/resnet_model/conv2d_9/Conv2D': [0.328125, 0.328125], 'model/resnet_model/conv2d_27/Conv2D': [0.2734375, 1.0], 'model/resnet_model/conv2d_52/Conv2D': [1, 1.0], 'model/resnet_model/conv2d_1/Conv2D': [0.296875, 1.0], 'model/resnet_model/conv2d_15/Conv2D': [0.3046875, 0.296875], 'model/resnet_model/conv2d_28/Conv2D': [0.27734375, 0.265625], 'model/resnet_model/conv2d_40/Conv2D': [0.3525390625, 0.26171875], 'model/resnet_model/conv2d_14/Conv2D': [0.2734375, 1.0], 'model/resnet_model/conv2d_26/Conv2D': [0.28125, 0.2734375], 'model/resnet_model/conv2d_20/Conv2D': [0.28125, 1.0], 'model/resnet_model/conv2d_42/Conv2D': [0.28515625, 1.0], 'model/resnet_model/conv2d_2/Conv2D': [0.34375, 0.296875], 'model/resnet_model/conv2d_34/Conv2D': [0.3095703125, 0.26171875], 'model/resnet_model/conv2d_22/Conv2D': [0.3046875, 0.28125], 'model/resnet_model/conv2d_17/Conv2D': [0.28125, 1.0], 'model/resnet_model/conv2d_29/Conv2D': [0.265625, 0.2734375]}
INFO:tensorflow:Actural preserv ratio: 1
INFO:tensorflow:roll-out #0: a-loss = 0.00e+00 | c-loss = 0.00e+00 | noise std. = 1.00e+00
INFO:tensorflow:best reward updated: -inf -> 0.1547
INFO:tensorflow:The best pruned model occured with
                strategy: [1.0, 0.296875, 0.34375, 0.296875, 0.296875, 0.328125, 0.3125, 0.3125, 0.34765625, 0.328125, 0.328125, 0.27734375, 0.33984375, 0.28125, 0.2734375, 0.3046875, 0.296875, 0.28125, 0.30078125, 0.2890625, 0.28125, 0.3125, 0.3046875, 0.28125, 0.271484375, 0.28515625, 0.28125, 0.2734375, 0.27734375, 0.265625, 0.2734375, 0.2978515625, 0.26953125, 0.2734375, 0.3095703125, 0.26171875, 0.28515625, 0.326171875, 0.26171875, 0.28515625, 0.3525390625, 0.26171875, 0.28515625, 0.283203125, 0.2783203125, 0.271484375, 0.298828125, 0.318359375, 0.2734375, 0.294921875, 0.328125, 0.265625, 1.0],
                accuracy: 0.15468749403953552 and
                pruned ratio: 737177119.0

**Comment by jiaxiang-wu** [comment]
@psyyz10 Could you please take a look at this? It seems to be related to `ChannelPrunedLearner`.

**Comment by xiaomr** [comment]
My system info is GPU 1080 Ti,  my  command is    sh  scripts/run_docker.sh nets/resnet_at_ilsvrc12_run.py   --resnet_size 50  --learner channel -n=4,  I tried several times . is it normal when the reward -inf  appear?
INFO:tensorflow:best reward updated: -inf -> 0.1547

**Comment by jiaxiang-wu** [comment]
@xiaomr 
1. It is normal that reward "-inf" appear. It is initialized as "-inf" and then immediately updated in the first roll-out, as you posted.
2. Does the training hanging problem consistently exist? Have you tried to re-run it?

**Comment by jiaxiang-wu** [comment]
@xiaomr Any updates?
